main.go

```diff
@@ -77,39 +96,62 @@ func getRawDiskUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmName
 			}
 		}
 	}
-	return "", fmt.Errorf("Could not get raw disk URL from the VirtualMachineExport object.")
+	return "", fmt.Errorf("Volume %s is not found in VirtualMachineExport internal volumes", volumeName)
 }
 
-func getVirtualMachineExportOnceReady(client kubecli.KubevirtClient, vmNamespace, vmName string) (*v1beta1.VirtualMachineExport, error) {
-	var vmExport *v1beta1.VirtualMachineExport
-
+func waitUntilVirtualMachineExportReady(client kubecli.KubevirtClient, vmNamespace, vmName string) error {
 	poller := func(ctx context.Context) (bool, error) {
 		vmExport, err := client.VirtualMachineExport(vmNamespace).Get(ctx, vmName, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
 
-		if vmExport.Status.Phase == v1beta1.Ready {
+		if vmExport.Status != nil && vmExport.Status.Phase == v1beta1.Ready {
 			return true, nil
 		}
 		return false, nil
 	}
 
-	err := wait.PollUntilContextTimeout(context.Background(), pollInterval, pollTimeout, true, poller)
+	return wait.PollUntilContextTimeout(context.Background(), pollInterval, pollTimeout, true, poller)
+}
+
+func getCertificateFromVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName string) (string, error) {
+	vmExport, err := client.VirtualMachineExport(vmNamespace).Get(context.Background(), vmName, metav1.GetOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("Failed to wait for VirtualMachineExport to be ready: %v", err)
+		return "", err
+	}
+
+	if vmExport.Status.Links == nil && vmExport.Status.Links.Internal == nil {
+		return "", fmt.Errorf("No links found in VirtualMachineExport status")
+	}
+
+	content := vmExport.Status.Links.Internal.Cert
+	if content == "" {
+		return "", fmt.Errorf("No certificate found in VirtualMachineExport status")
 	}
-	return vmExport, nil
+	return content, nil
 }
 
-func convertRawDiskImageToQcow2(rawDiskUrl string) error {
-	log.Println("Converting raw disk image to qcow2 format...")
+func getExportToken(client kubecli.KubevirtClient, vmNamespace, vmName string) (string, error) {
+	secret, err := client.CoreV1().Secrets(vmNamespace).Get(context.Background(), vmName, metav1.GetOptions{})
```

Comment:
why do you here retrieving the secret from api server, when you already have it in code available?

Reply:
but in this case the data in secret should not be modified by anyone, no?

Reply:
> but in this case the data in secret should not be modified by anyone, no?

I don't know, but I know that it should be readable by VMExport if I'm not mistaken. Let's see how it looks like after code refactoring (reducing to packages).